packages/tests/server/validators.test.ts

```diff
@@ -221,6 +223,57 @@ test('myzod', async () => {
   await close();
 });
 
+test('arktype schema', async () => {
+  const t = initTRPC.create();
+
+  const router = t.router({
+    num: t.procedure
+      .input(arktype.type({ text: 'string' }).assert)
+      .query(({ input }) => {
+        expectTypeOf(input).toMatchTypeOf<{ text: string }>();
+        return {
+          input,
+        };
+      }),
+  });
+
+  const { close, proxy } = routerToServerAndClientNew(router);
+  const res = await proxy.num.query({ text: '123' });
+  expect(res.input).toMatchObject({ text: '123' });
+
+  // @ts-expect-error this only accepts a `number`
+  await expect(proxy.num.query('13')).rejects.toMatchInlineSnapshot(`
+	[TRPCClientError: Must be an object (was string)]
+`);
+  await close();
+});
+
+test('effect schema', async () => {
+  const t = initTRPC.create();
+
+  const router = t.router({
+    num: t.procedure
+      .input(S.parse(S.struct({ text: S.string })))
+      .query(({ input }) => {
+        expectTypeOf(input).toMatchTypeOf<{ text: string }>();
+        return {
+          input,
+        };
+      }),
+  });
+
+  const { close, proxy } = routerToServerAndClientNew(router);
+  const res = await proxy.num.query({ text: '123' });
+  expect(res.input).toMatchObject({ text: '123' });
+
+  // @ts-expect-error this only accepts a `number`
+  await expect(proxy.num.query('13')).rejects.toMatchInlineSnapshot(`
+	[TRPCClientError: error(s) found
+	└─ Expected a generic object, actual "13"]
+`);
+  await close();
+});
+
```

Comment:
I wanted to add typia here, but its setup is [a lot more involved](https://typia.io/docs/setup/#generation). Interested in thoughts on that as it might add complexity to running these tests but may be fine

Reply:
#won't-fix then :)